version: "3.9"

services:
  django_app_with_gunicorn:
    image: registry.gitlab.com/toobusynow/dt.backend.bank-bot/dev:lates

    env_file: 
      - .env.docker.server
    container_name: django_gunicorn

    expose:
      - ${DJANGO_PORT}

    volumes:
      - static_volume:/backend/static/

    depends_on:
      db:
        condition: service_healthy

  db:
    image: postgres:15-alpine3.17
    container_name: postgres_db

    env_file: 
      - .env.docker.server
    volumes:
      - postgres_data:/var/lib/postgresql/data/

    environment:
      - POSTGRES_USER=${SQL_USER}
      - POSTGRES_PASSWORD=${SQL_PASSWORD}
      - POSTGRES_DB=${SQL_DATABASE}

    healthcheck:
      test: "pg_isready -q -h db"
      interval: 3s
      timeout: 5s
      retries: 5

    ports:
      - "5432:5432"

  nginx:
    build: ./nginx-static
    container_name: nginx_static
    ports:
      - 1337:80
    depends_on:
      - django_app_with_gunicorn
      
    volumes:
      - static_volume:/backend/static/
    
volumes:
  postgres_data:
  static_volume: